src/components/application_manager/src/commands/mobile/subscribe_way_points_request.cc

```diff
@@ -0,0 +1,75 @@
+#include "application_manager/application_manager_impl.h"
+#include "application_manager/commands/mobile/subscribe_way_points_request.h"
+
+namespace application_manager {
+
+namespace commands {
+
+SubscribeWayPointsRequest::SubscribeWayPointsRequest(
+    const MessageSharedPtr &message)
+    : CommandRequestImpl(message) {}
+
+SubscribeWayPointsRequest::~SubscribeWayPointsRequest() {}
+
+void SubscribeWayPointsRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app =
+      application_manager::ApplicationManagerImpl::instance()->application(
+          connection_key());
+
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "An application with connection key "
+                               << connection_key() << " is not registered.");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+
+  if (application_manager::ApplicationManagerImpl::instance()
+          ->IsAppSubscribedForWayPoints(app->app_id())) {
+    SendResponse(false, mobile_apis::Result::IGNORED);
+    return;
+  }
+
+  if (application_manager::ApplicationManagerImpl::instance()
+          ->IsAnyAppSubscribedForWayPoints()) {
+    application_manager::ApplicationManagerImpl::instance()
+        ->SubscribeAppForWayPoints(app->app_id());
```

Comment:
I was thinking about the ability to have the subscirprions with the application itself.
Right now I can see that you got to app manager to get an application and then go to the appmanager again to get the subscribtions which looks like overhead for me